### PR TITLE
Update utils to typescript

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed changes:

* Update the utils functions to use Typescript

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated commeP2nt below with a script to run)?

## Jetpack product discussion

P2: p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Go to `/wp-admin/admin.php?page=my-jetpack#/` and make sure the "Your plans" section works as expected (the utils functions are essentially only used here, outside of one being used on the golden token page)
image

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/data/types.ts [hunk 1]
```diff
@@ -36,3 +36,74 @@ export type WP_Error = {
 		status: number;
 	};
 };
+
+export type Purchase = {
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/_inc/data/types.ts:40]: @robertsreberski looks like you were right that we should've kept this 😅

Comment (robertsreberski) [reply projects/packages/my-jetpack/_inc/data/types.ts:40]: 😁

File: projects/packages/my-jetpack/_inc/utils/is-lifetime-purchase.ts [hunk 2]
```diff
@@ -1,26 +1,28 @@
+import type { Purchase } from '../data/types';
+
 /**
  * Check if a purchase is considered "Lifetime".
  *
  * @param {object} purchase - A WPCOM purchase object.
```
Comment (robertsreberski) [inline projects/packages/my-jetpack/_inc/utils/is-lifetime-purchase.ts]: You can actually use typescript types here ✨ 
```suggestion
 * @param {Purchase} purchase - A WPCOM purchase object.
```

**Review by robertsreberski — DISMISSED** [review]
One suggestion to JSDoc, nothing blocking

**Review by robertsreberski — APPROVED** [review]

**Pull request merged**